### PR TITLE
Autorun functionality

### PR DESCRIPTION
**Pull request by malleoz**
Does the following:
- Add Load/Scripts/Modules folder and add to search path
- Autorun scripts in Load/Scripts folder with _ prefix
- Don't autorun scripts in modules folder
- Prevent script execution when game not running
- Start script execution on game start, when toggled on
- End script execution on game end, if script running

**PLEASE test this before merging as I haven't had sufficient time.**

### CODE REVIEW

**Comment by gitrinitas** [comment]
merging w/ bypass branch protection now :2010:

File: Source/Core/Scripting/ScriptList.cpp
Comment (vabold) [inline Source/Core/Scripting/ScriptList.cpp:35]: The variable is called `m_scripts` despite no longer being a member of any struct or class. Perhaps `s_scripts` or `g_scripts` is preferable here instead.

A similar prefix would be recommended on `scripts_started`.

File: Source/Core/DolphinQt/Scripting/ScriptsListModel.cpp
Comment (malleoz) [reply Source/Core/DolphinQt/Scripting/ScriptsListModel.cpp:8]: Dolphin seems to love alphabetizing directory names, so I'm gonna swap line 7 with line 9, but otherwise yeah

File: Source/Core/Scripting/ScriptList.cpp
Comment (malleoz) [reply Source/Core/Scripting/ScriptList.cpp]: Went for g prefix. s_ indicates static which isn't really correct here.

**Review by vabold — APPROVED** [review]
Code looks good - DO NOT merge without testing.

**Comment by ximk** [comment]
We should consider whether modules should be able to be loaded at all, but this does as advertised. I have no issues to report, and this PR is ready to merge after we discuss this.

**Pull request merged**
